compiler/front/cli_reporter.nim

```diff
@@ -3700,6 +3700,12 @@ func astDiagToLegacyReport(conf: ConfigRef, diag: PAstDiag): Report {.inline.} =
       kind: rsemExpectedOrdinal,
       ast: diag.wrongNode,
       typ: diag.nonOrdTyp)
+  of adSemForExpectedIterator:
+    semRep = SemReport(
+      location: some diag.location,
+      reportInst: diag.instLoc.toReportLineInfo,
+      kind: rsemForExpectsIterator,
+      ast: diag.wrongNode)
```

Comment:
```suggestion
  of adSemForExpectedIterator:
    semRep = SemReport(
      location: some diag.location,
      reportInst: diag.instLoc.toReportLineInfo,
      kind: rsemForExpectsIterator,
      ast: diag.wrongNode)
```
This is unnecessary, as there's no special handling required for the diagnostic. Adding the `adSemForExpectedIterator` enum to the `of` branch in line 3293 is enough.